### PR TITLE
sort leaderboard table, fixes #229

### LINKED ISSUE #229: Leaderboard not sorted by metric value
The leaderboard tables in the output leaderboard.csv and the README.md are not sorted by the metric value. It will be convenient to compare and review our models if they are sorted.

Is there a way to identify if the metric is `greater_is_better` from this [method](https://github.com/mljar/mljar-supervised/blob/821500de420a35700f10d0a3126e01ab9d6fffc0/supervised/base_automl.py#L186)?

### PR DESCRIPTION
**Pull request by SuryaThiru**
Fixes the issue with the leaderboard not sorted by the metric value (#229).

### CODE REVIEW

**Comment by SuryaThiru** [comment]
Are there any other updates required for the PR?

**Comment by pplonski** [comment]
Hi @SuryaThiru! One comment for this PR. Could you use `get_minimize_direction()` in `base_automl.py` instead of checking it in the `get_leaderboard()`? By doing this, we will keep metric logic in the `Metric` class. Thank you!

**Comment by SuryaThiru** [comment]
Alright, currently the `Metric` class is imported but not used within `base_automl.py`. I will replace the string in `_eval_metric` with a `Metric` object.

**Comment by pplonski** [comment]
No no, the `_eval_mertic` is the string, it is set as input argument, please do not change it.

The `m` variable in `get_leaderboard()` is `Metric` class object. So you can just call `m.get_minimize_direction()`.

**Comment by SuryaThiru** [comment]
Oh, my bad. I assumed they were model objects. Thanks for the heads up, I'll make the changes.

**Comment by SuryaThiru** [comment]
The `m` variable turned out to be an object of `ModelFramework`/`Ensemble` class. So, I've temporarily added a `get_metric()` method to both the classes. Let me know if that would work for you

File: supervised/ensemble.py [hunk 1]
```diff
@@ -73,6 +73,9 @@ def get_name(self):
     def get_metric_name(self):
         return self.metric.name
 
+    def get_metric(self):
+        return Metric({'name': self.get_metric_name()})
```
Comment (pplonski) [inline supervised/ensemble.py:77]: No need to construct new object. The `self.metric` is already a `Metric` object.

File: supervised/model_framework.py [hunk 2]
```diff
@@ -202,6 +203,9 @@ def get_metric_name(self):
         self.metric_name = early_stopping.metric.name
         return early_stopping.metric.name
 
+    def get_metric(self):
+        return Metric({'name': self.get_metric_name()})
```
Comment (pplonski) [inline supervised/model_framework.py:207]: Similar like in the ensemble.

**Review by pplonski — CHANGES_REQUESTED** [review]
Please see the comments. Thank you!

File: supervised/model_framework.py (hunk 2 above)
Comment (SuryaThiru) [reply supervised/model_framework.py:207]: I don't think the `ModelFramework` has a `self.metric` attribute, if early stopping is used it is available via `early_stopping.metric`.  So I could do this:
```python 
def get_metric(self):
    early_stopping = self.callbacks.get("early_stopping")
    if early_stopping:
        return early_stopping.metric
    return Metric({'name': self.get_metric_name()})
```

Am I missing something?

Comment (pplonski) [reply supervised/model_framework.py:207]: Makes sense!

**Comment by SuryaThiru** [comment]
I think I've made the necessary changes. Let me know if something is missing.

**Comment by pplonski** [comment]
Hi @SuryaThiru! All changes are good. Thank you for your contribution. The PR will go into the next major release (`0.8.0`).

**Comment by pplonski** [comment]
@SuryaThiru thank you for your contribution! Changes will go to `0.8.0`

**Pull request merged**